src/metabase/models/collection.clj

```diff
@@ -871,14 +871,25 @@
   (let [collection (if (integer? collection-or-id)
                      (db/select-one [Collection :id :namespace] :id (collection-or-id))
                      collection-or-id)]
-    ;; HACK Collections in the "snippets" namespace have no-op permissions unless EE enhancements are enabled
-    ;;
-    ;; TODO -- Pretty sure snippet perms should be feature flagged by `advanced-permissions` instead
-    (if (and (= (u/qualified-name (:namespace collection)) "snippets")
-             (not (premium-features/enable-enhancements?)))
+    (cond
+      ;; HACK Collections in the "snippets" namespace have no-op permissions unless EE enhancements are enabled
+      ;;
+      ;; TODO -- Pretty sure snippet perms should be feature flagged by `advanced-permissions` instead
+      (and (= (u/qualified-name (:namespace collection)) "snippets")
+           (not (premium-features/enable-enhancements?)))
       #{}
+
+      ;; Collections in the "apps" namespace use the permissions of the root unless advanced permissions are enabled
+      (and (= (u/qualified-name (:namespace collection)) "apps")
+           (not (premium-features/has-feature? :advanced-permissions)))
+      #{(let [root (assoc root-collection :namespace :apps)]
+          (case read-or-write
+            :read  (perms/collection-read-path root)
+            :write (perms/collection-readwrite-path root)))}
```

Comment:
The intention is to check the permission on the root app collection instead of the collection itself if the advanced permission feature is not enabled.

Reply:
Oh, I see. Gotcha.